astropy/time/core.py

```diff
@@ -1013,9 +1135,17 @@ def argmax(self, axis=None, out=None):
         """
         # For procedure, see comment on argmin.
         jd = self.jd1 + self.jd2
-        approx = jd.max(axis, keepdims=True)
+
+        if NUMPY_LT_1_11_2:
+            # MaskedArray.max ignores keepdims so do it by hand (numpy <= 1.10)
+            approx = np.max(jd, axis)
+            if axis is not None:
+                approx = np.expand_dims(approx, axis)
+        else:
+            approx = np.max(jd, axis, keepdims=True)
```

Comment:
Note that for 3.1 we are dropping Numpy < 1.13 so these kinds of blocks can be simplified. However https://github.com/astropy/astropy/pull/7058 will need a little work before being merged, so we can always just merge this as-is then remove later.